jsonschema/validate_test.go

```diff
@@ -259,6 +262,128 @@ func TestStructInstance(t *testing.T) {
 	}
 }
 
+func TestStructEmbedding(t *testing.T) {
+	// For basic flattening
+	type Foo struct {
+		ID   string `json:"id"`
+		Name string `json:"name"`
+		Type string `json:"type"`
+	}
+	type Bar struct {
+		*Foo         // Pointer embedded - should flatten.
+		Extra string `json:"extra"`
+	}
+
+	// For outer field shadowing a pointer embed
+	type Inner struct {
+		Conflict  string `json:"conflict_field"` // This string field should be ignored.
+		InnerOnly string `json:"inner_only"`
+	}
+	type Outer struct {
+		*Inner
```

Comment:
Test non-pointer embedding, and non-pointer unexported embedding.

Reply:
Done